python/cudf/cudf/core/column/numerical.py

```diff
@@ -403,17 +435,16 @@ def find_and_replace(
             replacement_col = _normalize_find_and_replace_input(
                 self.dtype, replacement
             )
-        replaced = self.copy()
         if len(replacement_col) == 1 and len(to_replace_col) > 1:
             replacement_col = column.as_column(
                 utils.scalar_broadcast_to(
                     replacement[0], (len(to_replace_col),), self.dtype
                 )
             )
         elif len(replacement_col) == 1 and len(to_replace_col) == 0:
-            return replaced
+            return self.copy()
         to_replace_col, replacement_col, replaced = numeric_normalize_types(
-            to_replace_col, replacement_col, replaced
+            to_replace_col, replacement_col, self
         )
         df = cudf.DataFrame({"old": to_replace_col, "new": replacement_col})
```

Comment:
Following the changes elsewhere in this PR, I think this can be updated as well:
```suggestion
        df = cudf.DataFrame._from_data(
            {"old": to_replace_col, "new": replacement_col}
        )
```

Reply:
Yup, missed it